crates/vhost-user-backend/src/handler.rs

```diff
@@ -349,6 +349,14 @@ where
             self.vrings[index as usize]
                 .set_queue_info(desc_table, avail_ring, used_ring)
                 .map_err(|_| VhostUserError::InvalidParam)?;
+
+            // SET_VRING_BASE will only restore the avail index, so we need to fetch the used index
+            // from the vring as set by the guest
```

Comment:
I would suggest adding some more info included in the commit description here as well, such as the behavior of QEMU

Reply:
Ok